rfcs/convergence/file-naming.md

```diff
@@ -0,0 +1,152 @@
+| storybook (NONE)    | example.stories.tsx | `export const Example = () => {..}`                                                 | Specific file type for Storybook consumption. Main "starting" point for inner loop - daily development.                                                                                                                                                                                                                   |
+| mdx (NONE)          | example.stories.mdx | ---                                                                                 | Markdown(MDX) for consumer facing documentation(mainly for storybook at the moment, but can be transformed to any other tool that can parse MDX).                                                                                                                                                                         |
+| renderExample.tsx   | example.view.tsx    | `export const render = () => {..}`                                                  | `.render` might work as well, to match symbol name.                                                                                                                                                                                                                                                                       |
+| useExample.ts       | example.hook.ts     | `export const useExample = () => {}`                                                | main logic encapsulated under hook -> name should be the same as encapsulated component (in our case `example`)                                                                                                                                                                                                           |
+| useExampleStyles.ts | example.styles.ts   | `export const useStyles = makeStyles(); export const useIconStyles = makeStyles();` | with `makeStyles` 1 to N hooks for particular component styling needs. Why we currently use a solution that provides hooks, having `.styles` could be a good enough convention to explicitly communicate the intent. (if we switch implementation in the future we wouldn't need to change the type name as it's generic) |
+| useExampleRandom.ts | example.utils.ts    | `export const foo = () => {}`                                                       | aggregated set of reusable utils used within other parts of component implementation. Note there are no real react hooks!                                                                                                                                                                                                 |
```

Comment:
we have something like this: https://github.com/microsoft/fluentui/tree/master/packages/react-avatar/src/components/utils

I added it rather for demonstration if such a case would occur. I can remove it from the RFC if you think it might be confusing ?

Reply:
Don't think it's confusing at all, mostly just not a fan of utils 😄. I think it's best if we keep it for edge cases but mark it as optional, what do you think?

Reply:
> mark it as optional

what do you mean ?

Reply:
My intention was to somehow make sure everyone is aware .utils files are optional, to prevent the idea that one would need all these files regardless of implementation.